app/api/routes/user.py

```diff
@@ -37,3 +47,166 @@ async def create_user(user: UserCreateRequestModel):
+    return UserUpdateResponseModel(
+        id=user.id,
+        username=user.username,
+        email=user.email,
+        is_admin=user.is_admin,
+        is_active=user.is_active,
+        created_at=user.created_at,
+        updated_at=user.updated_at
+    )
+
+
+
+
+#UPDATE user detail
+@router.put("/users/{user_id}", response_model=UserUpdateResponseModel)
+async def update_user(user_id: UUID, update_data: UserUpdateRequestModel, current_user: User = Depends(get_current_active_user)):
+
+    #Authenticated user is trying to update their own information
+    if str(user_id) != str(current_user.id):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="You can only update your own details.")
+    
+    user = users_db.get(str(user_id))
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found.")
+    
+    #Validate and update the email if provided
+    if update_data.email:
+        if any(existing_user.email == update_data.email for existing_user in users_db.values() if existing_user.id != user_id):
+            raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Email already registered.")
+        user.email = update_data.email
+
+    if update_data.username:
+        user.username = update_data.username
+
+    if update_data.password:
+        if not is_valid_password(update_data.password):
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Password does not meet the security requirements.")
+        user.hashed_password = get_password_hash(update_data.password)
+
+    user.updated_at = datetime.now(timezone.utc)
+
+    users_db[str(user.id)] = user
+
+    return UserUpdateResponseModel(
+        id=user.id,
+        username=user.username,
+        email=user.email,
+        is_admin=user.is_admin,
+        is_active=user.is_active,
+        created_at=user.created_at,
+        updated_at=user.updated_at
+    )
+
+
+
+#DELETE User
+@router.delete("/users/{user_id}", status_code=status.HTTP_204_NO_CONTENT)
+async def delete_user(user_id: UUID, current_user: User = Depends(get_current_user)):
+    
+    if str(user_id) != str(current_user.id):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="You can only delete your own account.")
+
+    user = users_db.get(str(user_id))
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found.")
+
+    """
+    This part will be for checking active orders
+    """
+
+    del users_db[str(user_id)]
+    #Return 204 no content
+    return
+
+
+#GET all users 
+@router.get("/users", response_model=list[UserCreateResponseModel], status_code=status.HTTP_200_OK)
+async def get_all_users(current_admin: User = Depends(get_current_admin)):
+
+    try:
+        all_users = list(users_db.values())
+
+        return [
+            UserUpdateResponseModel(
```

Comment:
UserUpdateResponseModel !!!!

Reply:
I think there are some endpoints that need to include the modification time in the output, as well as which user was modified at that time, I’ve used the same model since it already provides the required fields. This approach is captured effectively.

Reply:
It's not a good practice to share models between different endpoints like that. If there is a common functionality, move it to a parent class. However, we should never return an updated model in a GET method. In your case, you should return a list of the same model type for the GET endpoint:
```python
List[GetUserResponseModel]
```

Reply:
Done.